### PR TITLE
Data prep

### PR DESCRIPTION
**Pull request by sallamander**
@dansbecker Here's what I've got so far. I'm a huge fan of trying to make things easy for people to follow along with and reproduce, but will defer to you here. At the very least, the `parse_xml.sh` in the `code/data_setup` folder is probably something we'll want.

### CODE REVIEW

File: data/mk_folders.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
```
Comment (dansbecker) [inline data/mk_folders.sh]: I usually expect the data folder to contain nothing but the data.  So, I wouldn't have expected to look there if I were looking for code (and I might expect to be able to iterate over the directory in some circumstances).

In theory, you could move this file to another directory... but it'd be more common to put stuff like these 4 lines directly in the Makefile underneath the appropriate targets.

File: .gitignore [hunk 2]
```diff
@@ -60,3 +60,8 @@ target/
 
 #Ipython Notebook
 .ipynb_checkpoints
+
```
Comment (dansbecker) [inline .gitignore:63]: You ok with just putting /data here.

I don't think anything in /data should be in source control

As a convention, I like to put raw data as we receive it in /data and intermediate data steps in /work

Are you ok with this convention?  If so, we can put both /data and /work in .gitignore.

**Comment by dansbecker** [comment]
That Makefile is great.  Nice!

File: data/mk_folders.sh (hunk 1 above)
Comment (sallamander) [inline data/mk_folders.sh]: Right on, that makes total sense. I can move it to the Makefile and adjust that there.

File: .gitignore (hunk 2 above)
Comment (sallamander) [inline .gitignore:63]: Yep, sounds great!

**Comment by sallamander** [comment]
So, what's the best strategy in terms of merging this all in? Should I adjust the `Makefile` with the changes we discussed, or is it best to let one person handle it all so merge conflicts don't creep in?

Happy to do whatever. The only person I've really collaborated with on a somewhat VC'd repo where we had lots of branches was Cary, so I'll be learning as we go (but promise not to blow anything up).

**Comment by dansbecker** [comment]
Yeah.  In general, either of us should feel free to merge anything the
other person has reviewed.  Whoever merges second may get a conflict on
.gitignore which will be easy to resolve.  That should be the only conflict
in the near term.

We won't get a conflict in the makefile, because I'm not going to touch it
until your PR is merged to master, and then I'll create a new branch off
that to update the Makefile.  In general, short but frequent PR's tend to
minimize merge conflicts.

**Comment by sallamander** [comment]
Okay, awesome. Merging now. I left the folder structure for data largely the, but just removed the `raw_data` intermediary, and moved anything that wasn't a direct download into the `work` folder (with `ILSVRC` and `alov` subfolders). If I misunderstood the structure we wanted, feel free to change or let me know and I can. 

So, the two `csvs` that are created from parsing the XML's for training/validation will now end up in `work/ILSVRC`. Next steps for me after merging your stuff in is to create a similar bash script to parse the videos and grab the frames for which we have bounding boxes.

**Pull request merged**